src/bulletproof/generators.rs

```diff
@@ -0,0 +1,276 @@
+use curve25519_dalek::edwards::{CompressedEdwardsY, EdwardsPoint};
+use curve25519_dalek::scalar::Scalar;
+use curve25519_dalek::traits::MultiscalarMul;
+use hash_edwards_to_edwards::hash_to_point;
+use tiny_keccak::{Hasher, Keccak};
+
+lazy_static::lazy_static! {
+    /// Alternate generator of ed25519.
+    ///
+    /// Obtained by hashing `curve25519_dalek::constants::ED25519_BASEPOINT_POINT`.
+    /// Originally used in Monero Ring Confidential Transactions.
+    pub static ref H: EdwardsPoint = {
+        CompressedEdwardsY(hex_literal::hex!(
+            "8b655970153799af2aeadc9ff1add0ea6c7251d54154cfa92c173a0dd39c1f94"
+        ))
+        .decompress()
+        .expect("edwards point")
+    };
+}
+
+/// Represents a pair of base points for Pedersen commitments.
+///
+/// The Bulletproofs implementation and API is designed to support
+/// pluggable bases for Pedersen commitments, so that the choice of
+/// bases is not hard-coded.
+///
+/// The default generators are:
+///
+/// * `B`: Monero's `H` basepoint;
+/// * `B_blinding`: the ed25519 basepoint
+/// hash-to-group on input `B_bytes`.
+#[derive(Copy, Clone)]
+pub struct PedersenGens {
+    /// Base for the committed value
+    pub B: EdwardsPoint,
+    /// Base for the blinding factor
+    pub B_blinding: EdwardsPoint,
+}
+
+impl PedersenGens {
+    /// Creates a Pedersen commitment using the value scalar and a blinding factor.
+    pub fn commit(&self, value: Scalar, blinding: Scalar) -> EdwardsPoint {
+        EdwardsPoint::multiscalar_mul(&[value, blinding], &[self.B, self.B_blinding])
+    }
+}
+
+impl Default for PedersenGens {
+    fn default() -> Self {
+        PedersenGens {
+            B: *H,
+            B_blinding: ED25519_BASEPOINT_POINT,
+        }
+    }
+}
+
+/// The `BulletproofGens` struct contains all the generators needed
+/// for aggregating up to `m` range proofs of up to `n` bits each.
+#[derive(Clone, Debug)]
+pub struct BulletproofGens {
+    /// The maximum number of usable generators for each party.
+    pub gens_capacity: usize,
+    /// Number of values or parties
+    pub party_capacity: usize,
+    /// Precomputed \\(\mathbf G\\) generators for each party.
+    pub G_vec: Vec<Vec<EdwardsPoint>>,
+    /// Precomputed \\(\mathbf H\\) generators for each party.
+    pub H_vec: Vec<Vec<EdwardsPoint>>,
+}
+
+impl BulletproofGens {
+    /// Create a new `BulletproofGens` object.
+    ///
+    /// # Inputs
+    ///
+    /// * `gens_capacity` is the number of generators to precompute
+    ///    for each party.  For rangeproofs, it is sufficient to pass
+    ///    `64`, the maximum bitsize of the rangeproofs.
+    ///
+    /// * `party_capacity` is the maximum number of parties that can
+    ///    produce an aggregated proof.
+    pub fn new(gens_capacity: usize, party_capacity: usize) -> Self {
+        fn varint_to_bytes(n: usize) -> Vec<u8> {
+            use integer_encoding::VarInt;
+            n.encode_var_vec()
+        }
```

Comment:
What is the use of this function if we just delegate to one other function? :D

Reply:
Yep, no longer need this wrapper!